esmf_utils/UngriddedDims.F90

```diff
@@ -79,6 +80,18 @@ function new_UngriddedDims_arr(dim_specs) result(spec)
 
    end function new_UngriddedDims_arr
 
+   function new_UngriddedDims_extent_arr(extent_arr) result(spec)
+      type(UngriddedDims) :: spec
+      integer, intent(in) :: extent_arr(:)
+
+      integer :: i
+      type(UngriddedDim) :: dim_spec
+
+      do i = 1, size(extent_arr)
+         dim_spec = UngriddedDim(extent_arr(i))
+         call spec%dim_specs%push_back(dim_spec)
+      end do
+   end function new_UngriddedDims_extent_arr
```

Comment:
This does not support the scalar case, which is the most important.   Indeed, I'm tempted to disallow this shortcut for multiple ungridded dimensions, as names and such become so much more important when there is more than one.

Reply:
I am not sure if we need a specific constructor for the scalar case. The scalar case can use the current interface by creating an array, as in `UngriddedDims([scalar_case])`. But I can always add one - `new_UngriddedDims_extent`.

Or we can go back to what I had originally, i.e. move the logic to `MAPL_Generic::gridcomp_add_fieldspec`